api-client/APIClient.test.ts

```diff
@@ -0,0 +1,81 @@
+import { z } from "zod"
+import { createApiClient } from "./api-client"
+import { GenericApiClient } from "./api-endpoint-types"
+
+const ApiSchema = {
+  "/users/{userId}": {
```

Comment:
What if for the schema, we structured it like this:

```ts
{
    users: {
        path: "/users/{userId}",
        get: { /* Zod Schemas */ },
        patch: { /* Zod Schemas */ },
        // ...
    } 
}
```

We could then call into it like:

```ts
const resp = await apiClient.users.get({ /* Stuff */ })
```

I think this reads a lot better and would probably give us better lsp support (and avoids linter warnings).

Reply:
Sure, up to @tristan8chu and @nairoldi if they want to give unique names to the endpoints